sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -844,9 +931,51 @@ case class HashAggregateExec(
          |// update unsafe row buffer
          |${updateUnsafeRowBuffer.mkString("\n").trim}
        """.stripMargin
+    } else {
+      // We need to copy the aggregation row buffer to a local row first because each aggregate
```

Comment:
Same question as the non-grouping version, why is it required to load the aggregate buffers upfront here?